spec/controllers/user/swaps_controller_spec.rb

```diff
@@ -21,9 +23,10 @@
     let(:an_email) { double(:an_email) }
 
     before do
-      session[:user_id] = new_user.id
-      allow(User).to receive(:find_by_id).with(new_user.id)
-                       .and_return(new_user)
+      # Stub out authentication
+      allow(request.env["warden"]).to receive(:authenticate!).and_return(new_user)
```

Comment:
Cunning, see https://github.com/plataformatec/devise/wiki/How-To:-Stub-authentication-in-controller-specs